shared-libs/translation-utils/src/translation-utils.js

```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  loadTranslations: function(values) {
+    Object.keys(values).forEach(function(key) {
+      var value = values[key];
+      if(value.indexOf('${') > -1) {
+        var ids = value.match(/\${[^}\r\n]*}/g);
+        ids.forEach(function(id) {
+          var idKey = id.replace(/[${}]/g, '');
+          value = value.replace(id, values[idKey] || id);
+        });
+        values[key] = value;
```

Comment:
I think this could be a bit cleaner if you use regex replace, eg:

```js
const KEY_REGEX = /\${([^}\r\n]*)}/g; // Note the additional parens
module.exports = {
  loadTranslations: values => {
    Object.keys(values).forEach(key => {
      values[key] = values[key].replace(KEY_REGEX, (match, key) => values[key] || ('${' + id + '}'));
    });
    return values;
  }
};
```

I always like to pull out regexes as constants so they don't have to be compiled all the time, especially in loops like this one.

This slightly different regex uses parens to define a group so the regex itself removes the `${` and `{` so you don't have to. It also does nothing if there are no matches so you don't need the `indexOf` bit.